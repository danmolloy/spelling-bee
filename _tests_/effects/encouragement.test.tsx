import "@testing-library/jest-dom"
import { render, screen } from "@testing-library/react"
import Encouragement, { EncouragementProps } from "../../app/effects/encouragement"

const mockProps: EncouragementProps = {
  points: Math.ceil(Math.random() * 6),
  text: "Amazing!"
}

describe("<Encouragement", () => {
  beforeEach(() => {
    render(<Encouragement {...mockProps} />)
  })
  it("encouragement-div is in the document", () => {
    const encouragementDiv = screen.getByTestId("encouragement-div")
    expect(encouragementDiv).toBeInTheDocument()
  })
  it("shows expected message with points", () => {
    const encouragementDiv = screen.getByTestId("encouragement-div")
    expect(encouragementDiv.textContent).toMatch(mockProps.text)
  })
})


describe("<Encouragement", () => {
  const mockProps: EncouragementProps = {
    points: 6,
    text: "Hooray!"
  }
  beforeEach(() => {
    render(<Encouragement {...mockProps} />)
  })
  it("matches snapshot", () => {
    const encouragementDiv = screen.getByTestId("encouragement-div")
    expect(encouragementDiv).toMatchSnapshot()
  })
})