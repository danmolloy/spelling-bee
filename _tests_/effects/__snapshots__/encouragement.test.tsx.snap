// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Encouragement matches snapshot 1`] = `
<div
  class="z-50 self-center absolute mt-20 h-12 flex flex-row items-center"
  data-testid="encouragement-div"
>
  <div
    class="message px-2 py-1 h-8 flex items-center bg-white border rounded"
  >
    <p>
      Hooray!
    </p>
  </div>
  <div
    class="points ml-2 flex items-center font-bold"
  >
    + 
    6
  </div>
</div>
`;
